### ISSUE TITLE
Improve Axe accessibility tests to cover different possible UI states

### ISSUE THREAD

**Issue by matthieu-foucault** [bcgov/cas-ciip-portal#1818]
UI / HTML that is hidden from view (because it will be toggled visible when another state is reached) tends to also be hidden from screen readers.

We think this probably means it's also hidden from Axe; therefore, the test needs to run through various possible states in order to get a full picture of the accessibility violations.

## Limitation #1: Focus management for interactively shown/hidden elements
As we get deeper into testing interactive states, we will encounter some issues with compliant focus management when things are interactively hidden and shown. One example is below, although this won't necessarily be limited to modal windows:

We have some currently disabled interactive assertions (for example on the products admin page) showing non-compliant focus management for opened modal windows, which would need to be addressed before we can enable them.

When a modal window opens, react-bootstrap applies an aria-hidden attribute to the <div id="__next"> page container (ironically with consideration to accessibility). The DOM elements for the overlay and the modal dialog are appended to the page body as siblings to #__next. Though not 100% visually hidden under the semi-transparent overlay, it's meant to indicate that the user's focus of interaction is now in the modal. The violation is because BS should probably be applying tabindex="-1" along with aria-hidden to prevent semantically hidden content from being focused and interacted with.

Axe explanation of this violation:

"Fix all of the following:
Focusable content should be disabled or be removed from the DOM
Focusable content should have tabindex='-1' or be removed from the DOM"

Background info
Modal dialogs are supposed to manage DOM focus to stay within the dialog while open (read in detail here), and if Bootstrap goes so far as to apply aria-hidden to the rest of the page, I'd have expected it to include this part out of the box (BS accessibility disclaimer) but apparently it does not.

## Limitation #2: Seemingly inexplicable contrast violations
When our (currently disabled) interactive accessibility tests of opened modal windows run, there's an additional contrast violation for dialog content that is obviously compliant. This needs further investigation, but I'm wondering if it may be a limitation of the tool - there are some instances where Chrome Devtools for example cannot calculate contrast.
